docs/sc-memory/api/cpp/extended/agents/actions.md

```diff
@@ -0,0 +1,345 @@
+# **C++ Actions API**
+
+!!! warning
+    This documentation is correct for only versions of sc-machine that >= 0.10.0.
+--- 
+
+This API provides functionality to handle actions on C++.
+
+## **What is action?**
+
+All actions are processes. Any process is performed by some subject. Each action performed can be interpreted as a process of solving some task, i.e. as a process of achieving the given goal with the given conditions. Each action denotes some transformation carried out in the external environment or in the memory of some system.
+
+Like an agent, an action has a specification. This specification includes: class, arguments, state, result and others. The specification of an action is called a problem (task).
+
+## **ScAction**
+
+The sc-machine provides `ScAction` class to handle actions in sc-memory. You can't call constructors of this class, because they are private. Use `ScAgentContext` to create action with provided class or use the existed one.
+
+```cpp
+// Find action class and create action.
+...
+ScAction action = context.CreateAction(actionClassAddr);
+...
+```
+
+```cpp
+// Or find action and use it.
+...
+ScAction action = context.UseAction(actionAddr);
+...
+```
+
+### **GetClass**
+
+To get class of action use this method.
+
+```cpp
+...
+ScAddr const & actionClassAddr = action.GetClass();
+...
+```
+
+### **GetArgument**
+
+Actions can have arguments. Action arguments are some objects on which this action should be performed. If action has arguments in knowledge base, then you will get them. There are a couple of getters for this.
+
+```cpp
+...
+// Provide relation here if your argument has specific role in action. 
+ScAddr const & argAddr = action.GetArgument(ScKeynodes::rrel_1);
+// If there is no argument with such role, then `argAddr` will be empty.
+...
+```
+
+```cpp
+...
+// Or provide index of argument. 1 is equal to `ScKeynodes::rrel_1`,
+// 2 is equal to `ScKeynodes::rrel_2` and etc.
+ScAddr const & argAddr = action.GetArgument(1);
+// If there is no argument with such role, then `argAddr` will be empty.
+...
+```
+
+!!! warning
+    This method will throw `utils::ExceptionInvalidParams` if you provide invalid index for argument (for example, 0).
```

Comment:
0 looks like one and only case that throws this exception

Reply:
There is an upper limit of 20. I will specify it in the documentation.